### PR TITLE
Kwxm/costing/improve signature verification benchmarks (PLT-199)

### PR DESCRIPTION
**Pull request by kwxm**
This modifies the cost model benchmarks for the signature verification builtins to only use valid inputs.  They previously used totally random inputs, but it turns out that some of the functions return immediately if the inputs fail some (undocumented/unadvertised) basic format checking;  this made the modelling process a bit inaccurate.  The new benchmarks will (I hope) always measure worst-case behaviour, and that's let me tidy up the R modelling code as well. I re-ran the benchmarks for the signature verification functions and got similar results to the existing ones, except in the case of `verifyEd25519Signature`, where the costing function was underestimating costs a bit (see also PLT-7).

### CODE REVIEW

File: plutus-core/cost-model/budgeting-bench/Benchmarks/CryptoAndHashes.hs [hunk 1]
```diff
@@ -1,74 +1,112 @@
...
+              in (vkBytes, msg, sigBytes)
 
 benchVerifyEd25519Signature :: Benchmark
 benchVerifyEd25519Signature =
-    createThreeTermBuiltinBenchElementwise name [] pubkeys messages signatures
-           where name = VerifyEd25519Signature
-                 pubkeys    = listOfSizedByteStrings 50 32
-                 messages   = bigByteStrings seedA
-                 signatures = listOfSizedByteStrings 50 64
--- TODO: this seems suspicious.  The benchmark results seem to be pretty much
```
Comment (kwxm) [inline plutus-core/cost-model/budgeting-bench/Benchmarks/CryptoAndHashes.hs:41]: This is (finally) explained by the check [here](https://github.com/input-output-hk/cardano-crypto/blob/07397f0e50da97eaa0575d93bee7ac4b2b2576ec/cbits/ed25519/ed25519.c#L103
) that certain bits are zero, which means that if we supply random signatures then seven times out of eight it'll return almost immediately and not do the expensive full verification process (which is what we really want to measure).

File: plutus-core/cost-model/data/models.R [hunk 2]
```diff
@@ -531,8 +514,13 @@ modelFun <- function(path) {
         m <- lm(t ~ y_mem, filtered)
         adjustModel(m,fname)
     }
-    
 
+    ## All of the arguments of VerifyEcdsaSecp256k1Signature are of fixed size.
```
Comment (michaelpj) [inline plutus-core/cost-model/data/models.R:518]: wait, I thought you were implying that the model should be linear now?

Comment (michaelpj) [reply plutus-core/cost-model/data/models.R:518]: Or is that what's going to happen in the other PR?

Comment (kwxm) [reply plutus-core/cost-model/data/models.R:518]: That's talking about `EcdsaSecp256k1`. It's `Ed25519` that's problematic.

**Pull request merged**